.github/release-checklist.md

```diff
@@ -0,0 +1,63 @@
+# Template to use for release PRs from `develop` to `main`
+
+:warning: **DO NOT MERGE (YET)** :warning:
+
+**Please **do** add approvals if you agree as otherwise we won't be able to release.**
+
+PR for tracking changes for the x.x.x release. Target release date: **DOW MONTH DAY YEAR**.
+
+## Release checklist
+
+### General
+
+- [ ] Verify, and if necessary, update the allowed version ranges for various dependencies in the `composer.json` - PR #xxx
+- [ ] PHPCS: check if there have been releases since the last WordPressCS release and check through the changelog to see if there is anything WordPressCS could take advantage of - PR #xxx
+- [ ] PHPCSUtils: check if there have been releases since the last WordPressCS release and update WordPressCS code to take advantage of any new utilities - PR #xxx
+- [ ] PHPCSExtra: check if there have been releases since the last WordPressCS release and check through the changelog to see if there is anything WordPressCS could take advantage of - PR #xxx
+- [ ] Check if the minimum WP version property needs updating in `MinimumWPVersionTrait::$default_minimum_wp_version` and if so, action it - PR #xxx
+- [ ] Check if any of the list based sniffs need updating and if so, action it.
+    List based sniffs:
```

Comment:
Can these include links to the relevant files?

Reply:
They could, but I'm not sure of the relevance ?

By the time the release PR is pulled, these updates should already have been made and the most relevant links would be the PR links confirming the update(s) have been made and the checklist asks for those PR links (which can be replaced with _N/A_ if no update was necessary).

What am I missing ?

Reply:
> the most relevant links would be the PR links confirming the update(s) have been made

... _if_ updates were needed. I see this release checklist as something to create earlier on in the getting-ready-for-a-release period, when the PRs may not be done yet, and so a quick link to the relevant files would make sense. If these "updates should already have been made", because you have other ways of remembering to do them, then there's less benefit in including them in the checklist at all.

Happy to be disagreed with :-)

Reply:
Well, the checklist - in my mind - serves more as a reminder to double-check that everything is ready for a release.

Like for PHPCSUtils and PHPCSExtra (and Requests for that matter), where I've been using these type of checklists for a while, and have been doing rapid releases (release quickly & often), there's always something in the list when I'm getting ready to release which I'm happy to be reminded off, even though most things are done & dusted.

I can add the links, but even when they are there, I would personally never use them as the kind of updates needed to the files listed here, are not the kind of updates I'd ever make using the web-interface, if only as it's never just one file which needs updating - let's not forget that the tests also will need updates -, so what purpose do the links then serve in the list ? (other than give the misleading impression to newcomers that those would be simple updates which _can_ be made through the web-interface ?)

For me, the purpose of the list is to have the reminders and make sure we don't forget any of the many tasks in getting the release out. And on the off-chance that we would occasionally skip a task, that doing so has to be a deliberate choice, which is then annotated in the checklist or via a comment in the PR.

And yes, also happy to be disagreed with ;-). Discussion makes things better.